### DISCUSSION TITLE
compilation to wasm

### DISCUSSION THREAD

**Discussion by lovasoa** [actix/actix-web#3137] [Ideas] [answered]
This is a follow-up on #340 

Many modern "edge function" providers (*Cloudflare Workers, Deno Deploy, Vercel Edge Functions...*) allow hosting server-side code very cheaply, but at one condition: it has to be compiled to wasm (or written in js). This makes wasm a very attractive compilation target for web server authors. As a maintainer of [SQLPage](https://github.com/lovasoa/SQLpage), for instance, I'd love to be able to distribute it as a wasm binary.

Wasm compatibility would also allow running server-side request handling code directly on the client, which would open many interesting use cases, such as serving some requests directly as service workers.

Making wasm and wasi officially supported target would but actix web on par with other rust web frameworks such as axum.

**Comment by robjtede** [comment, marked as answer]
I have no capacity to look into this. As I said in that linked issue:

> For WASI support, I'd ask for a new and fleshed out feature request issue.
